app/models/mixins/relationship_mixin.rb

```diff
@@ -622,19 +622,18 @@ def add_children(*child_objs)
   alias_method :add_child, :add_children
 
   def parent=(parent)
-    if use_ancestry?
```

Comment:
i like this guard clause
granted it doesn't properly set the type when there are parents, but having the caller seems to be close to working with this.

are you coding it this way so we can set the type on the parent if it is not nil?

Reply:
yes

Reply:
do i still have to put it back the way it was?

Reply:
lookin' good. nice change